composer/utils/checkpoint.py

```diff
@@ -716,7 +773,7 @@ def safe_torch_load(
 
             return state_dict
         else:
-            return torch.load(composer_states_filepath, map_location=map_location)
+            return _torch_load_with_validation(composer_states_filepath, map_location=map_location)
```

Comment:
note that elastic sharded checkpoints don't use torch.load (they use `torch.distributed.checkpoint.load_state_dict`), so this validation function wouldn't be called for that

Reply:
where is the file load for sharded checkpoints? Doesn't it still go through `_restore_checkpoint` (which starts by loading the state dict into memory)

Reply:
ah nvm, i see it is `load_sharded_checkpoint`

Reply:
Do you know where we can inject this for sharded checkpoints? It needs to run before we load the file into memory. Maybe loop over folder before calling `load_state_dict`?